File: web/src/state/PlayerStatus.ts
import { ref, computed } from 'vue'
import { useNuiEvent } from '../utils/useNui'
import type { PlayerStatus } from '../types'

const status = ref<PlayerStatus>({
  health: 100,
  armour: 100,
  hunger: 100,
  thirst: 100,
  stamina: 100
})

export function usePlayerStatus() {
  useNuiEvent<PlayerStatus>('updatePlayerStatus', (data) => {
    status.value = data
  })

  return {
    status: computed(() => status.value)
  }
}

File: client/threads/PlayerStatus.lua
local PlayerStatus = {}


local Framework = require("client.framework." .. config.framework)

---@return boolean
function PlayerStatus:start()
    CreateThread(function()
        while true do
            local ped = cache.ped

            local health = GetEntityHealth(ped) - 100
            local armor = GetPedArmour(ped)
            local stamina = GetPlayerStamina(PlayerId())

            local hunger = Framework:GetPlayerStatus(ped, 'hunger')
            local thirst = Framework:GetPlayerStatus(ped, 'thirst')
        
            SendNUIMessage({
                action = 'updatePlayerStatus',
                data = {
                    health = health,
                    armour = armor,
                    stamina = stamina,
                    hunger = hunger,
                    thirst = thirst,
                }
            })

            Wait(800)
        end
    end)
    
    return true
end

return PlayerStatus